azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -1166,88 +1182,36 @@ private void handleJobHistoryPage(final HttpServletRequest req,
       return;
     }
 
-    final String jobId = getParam(req, "job");
-    final int pageNum = getIntParam(req, "page", 1);
-    final int pageSize = getIntParam(req, "size", 25);
-
     page.add("projectId", project.getId());
     page.add("projectName", project.getName());
-    page.add("jobid", jobId);
-    page.add("page", pageNum);
-
-    final int skipPage = (pageNum - 1) * pageSize;
 
-    int numResults = 0;
     try {
-      numResults = this.executorManagerAdapter.getNumberOfJobExecutions(project, jobId);
-      final int maxPage = (numResults / pageSize) + 1;
-      List<ExecutableJobInfo> jobInfo =
-          this.executorManagerAdapter.getExecutableJobs(project, jobId, skipPage, pageSize);
+      final int numResults = this.executorManagerAdapter.getNumberOfJobExecutions(project, jobId);
+      page.add("recordCount", numResults);
 
-      if (jobInfo == null || jobInfo.isEmpty()) {
-        jobInfo = null;
+      final int totalPages = ((numResults - 1) / pageSize) + 1;
+      if (pageNum > totalPages) {
+        pageNum = totalPages;
+        page.add("page", pageNum);
       }
-      page.add("history", jobInfo);
+      final int elementsToSkip = (pageNum - 1) * pageSize;
+      final List<ExecutableJobInfo> jobInfo =
+          this.executorManagerAdapter.getExecutableJobs(project, jobId, elementsToSkip, pageSize);
 
-      page.add("previous", new PageSelection("Previous", pageSize, true, false,
-          Math.max(pageNum - 1, 1)));
+      if (jobInfo != null && !jobInfo.isEmpty()) {
```

Comment:
consider this?: https://commons.apache.org/proper/commons-collections/apidocs/org/apache/commons/collections4/CollectionUtils.html#isNotEmpty-java.util.Collection-

Reply:
I changed the if condition to use the utility method isNotEmpty(). Thanks